JS/Lesson34/goksel/recipe-app/src/modules/user/RegistrationForm/index.tsx

```diff
@@ -0,0 +1,187 @@
+import { useForm } from "react-hook-form";
+import { Link, unstable_HistoryRouter } from 'react-router-dom';
+import Avatar from '@mui/material/Avatar';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import { Box, Container, Grid, TextField, Typography, FormControl, InputLabel, OutlinedInput, InputAdornment, IconButton, FormControlLabel, Button, Snackbar } from "@mui/material";
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import * as React from 'react';
+import Checkbox from '@mui/material/Checkbox';
+
+
+interface RegistrationFormProps {
+	open: boolean;
+	setOpen: any;
+	firstName: string;
+	lastName: string;
+	email:string;
+	password:string;
+}
+
+export const RegistrationForm = () => {
+
+	const [open, setOpen] =React.useState(false);
+
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+		reset
+	} = useForm<RegistrationFormProps>();
+
+
+	const onSubmit = (data:RegistrationFormProps) => {	
+		console.log(data);
+		reset();
+		setOpen(true);
+	}
+
+	const [showPassword, setShowPassword] =React.useState(false);
+
+	const handleClickShowPassword = () => setShowPassword(!showPassword);
+
+	return (
+
+		<Container component="main" maxWidth="xs">
+			<Box sx={{ display: "flex", flexDirection: "column", alignItems: "center" }}>
+				<Avatar sx={{ mt: 8, bgcolor: "primary.main" }}>
+					<LockOutlinedIcon />
+				</Avatar>
+				<Typography component="h1" variant='h5'>Sign Up</Typography>
+
+				<Box component="form" noValidate sx={{ mt: 3 }} onSubmit={handleSubmit(onSubmit)}>
+
+					<Grid container spacing={2}>
+						<Grid item xs={12} sm={6}>
+							<TextField
+								required
+								fullWidth
+								//@ts-ignore
+								name="firstName"
+								id="firstName"
+								label="First Name"
+								{...register("firstName")}
+							/>
+						</Grid>
+
+						<Grid item xs={12} sm={6}>
+							<TextField
+								required
+								fullWidth
+								//@ts-ignore
+								name="lastName"
+								id="lastName"
+								label="Last Name"
+								{...register("lastName")}
+							/>
+						</Grid>
+
+						<Grid item xs={12}>
+							<TextField
+								required
+								fullWidth
+								//@ts-ignore
+								name="email"
+								id="email"
+								label="Email Adress"
+								type="email"
+								error={!!errors.email}
+								helperText={errors.email ? errors.email.message : ""}
+								{...register("email", {
+									required: "User Email is Required",
+									pattern: {
+										value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
+										message: 'Please enter a valid email adress'
+									}
+								})}
+							/>
+						</Grid>
+
+						<Grid item xs={12}>
+							<FormControl
+								variant='outlined'
+								required
+								size='medium'
+								fullWidth
+								error={!!errors.password}
+								helperText={errors.password ? errors.password.message : ''}
```

Comment:
[FormControl](https://mui.com/material-ui/api/form-control/) element does not accept `helperText` prop. Instead, you need to use a `FormHelperText` component inside `FormControl` component.
E.g.
```

<FormControl>
  <InputLabel htmlFor="my-input">Email address</InputLabel>
  <Input id="my-input" aria-describedby="my-helper-text" />
  <FormHelperText id="my-helper-text">We'll never share your email.</FormHelperText>
</FormControl>
```

Reply:
Fixed